### PR TITLE
Update CMakeLists.txt

### PR DESCRIPTION
**Pull request by noeyjd**
@yucanding @Qicang Defined a default set of compiler flags for release, debug, and all builds. Although I am not sure if this is accurate, please review my work accordingly.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,16 +3,46 @@ set(CMAKE_CXX_STANDARD 11)
...
 
+list(APPEND flags "-fPIC" "-Wall")
+if(NOT WIN32)
+  list(APPEND flags "-Wextra" "-Wpedantic")
+endif()
+
+set(CMAKE_CXX_FLAGS         "${CMAKE_CXX_FLAGS} -Wall"
+set(CMAKE_CXX_FLAGS_DEBUG   "-g -traceback")
+set(CMAKE_CXX_FLAGS_RELEASE "-03 -ip xHOST")
+
```
Comment (yucanding) [inline CMakeLists.txt:27]: I think maybe we can set the default one (Release for example) in this way:
```suggestion
if(NOT CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE "Release")
endif()
```

**Review by yucanding — COMMENTED** [review]
Checks have failed so the current codes contain errors.

**Comment by Qicang** [comment]
The checks are not passed. Be sure to check why the checks failed. @noeyjd @yucanding When you change the CMakeLists, you should test it locally to see whether you can compile the code.

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,16 +3,46 @@ set(CMAKE_CXX_STANDARD 11)
...
 
 enable_testing()
 add_subdirectory(test)
 
+list(APPEND flags "-fPIC" "-Wall")
+if(NOT WIN32)
+  list(APPEND flags "-Wextra" "-Wpedantic")
+endif()
+
+set(CMAKE_CXX_FLAGS         "${CMAKE_CXX_FLAGS} -Wall"
```
Comment (Qicang) [inline CMakeLists.txt:24]: missing )

File: CMakeLists.txt [hunk 3]
```diff
@@ -3,16 +3,46 @@ set(CMAKE_CXX_STANDARD 11)
...
+
+target_compile_option(ners570_SpMV
+  PRIVATE
+    ${CXX_FLAGS}
+    "$<$<CONFIG:Debug>:${CXX_FLAGS_DEBUG}>"
+    "$<$<CONFIG:Release>:${CXX_FLAGS_RELEASE}>"
+  )
+
+$ make clean
+$ make VERBOSE=1
```
Comment (Qicang) [inline CMakeLists.txt:48]: What is the purpose of these two commands?

**Review by Qicang — CHANGES_REQUESTED** [review]

**Review by Qicang — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 4]
```diff
@@ -3,16 +3,46 @@ set(CMAKE_CXX_STANDARD 11)
...
+  list(APPEND CXX_FLAGS_DEBUG "-Wdocumentation")
+  list(APPEND CXX_FLAGS_RELEASE "-O3" "-Wno-unused")
+endif()
+
+target_compile_option(ners570_SpMV
+  PRIVATE
+    ${CXX_FLAGS}
+    "$<$<CONFIG:Debug>:${CXX_FLAGS_DEBUG}>"
+    "$<$<CONFIG:Release>:${CXX_FLAGS_RELEASE}>"
+  )
```
Comment (Qicang) [inline CMakeLists.txt:45]: Get rid of it

**Review by Qicang — CHANGES_REQUESTED** [review]

**Comment by yucanding** [comment]
Update CMakeLists.txt

**Pull request merged**